### PR TITLE
1111591 - require passwords to not be empty strings

### PR DESCRIPTION
**Pull request by asmacdo**
[BZ-1111591](https://bugzilla.redhat.com/show_bug.cgi?id=1111591)

Okaara cannot recognize an empty string as a password, so we should not allow the user to get into a broken state.

Reopened from https://github.com/pulp/pulp/pull/1367

### CODE REVIEW

**Comment by asmacdo** [comment]
@rbarlow, since you reviewed the WIP, could you take a look at this?

**Comment by rbarlow** [comment]
This PR needs some unit tests to go with the new change. Let me know when you've added a test to make sure the empty password is handled correctly, and I'll check it out again.

**Comment by asmacdo** [comment]
@rbarlow, there were very minimal tests of the affected functions. I have expanded these tests to cover all affected code.

File: client_admin/test/unit/test_auth.py [hunk 1]
```diff
@@ -55,29 +54,92 @@ def test_invalid_input(self):
...
-class TestAuthUser(base_builtins.PulpClientTests):
+class TestAuthUser(unittest.TestCase):
 
-    USER_LOGIN = 'test-login'
-    PASSWORD = 'test-password'
-    NAME = 'test-name'
+    def test__prompt_password_given(self):
+        """
+        Test that prompt password will ask for a password until the password
+        is not blank
```
Comment (rbarlow) [inline client_admin/test/unit/test_auth.py]: This docstring doesn't seem accurate for what this test is doing. It looks like the tests just makes sure the password works in the normal case.

**Comment by rbarlow** [comment]
LGTM.

**Pull request merged**